applications/FluidDynamicsApplication/custom_elements/alternative_d_vms_dem_coupled.cpp

```diff
@@ -0,0 +1,999 @@
+            // OSS pressure subscale projection
+            double DPhi = rData.DN_DX(i,d) * tau_two * fluid_fraction * (mass_source - fluid_fraction_rate - MassProj);
+            rLocalRHS[row+d] += rData.Weight * (VF - VI + AF + DPhi - RSigmaF + GBetaF - DBetaF + VPhi);
+        }
+        double Q = rData.N[i] * (mass_source - fluid_fraction_rate);
+        rLocalRHS[row+Dim] += rData.Weight * (QAlphaF + Q); // Grad(q) * TauOne * (Density * BodyForce)
+    }
+
+    // Write (the linearized part of the) local contribution into residual form (A*dx = b - A*x)
+    array_1d<double,LocalSize> values;
+    this->GetCurrentValuesVector(rData,values);
+    noalias(rLocalRHS) -= prod(LHS, values);
+
+    /* Viscous contribution (with symmetric gradient 2*nu*{E(u) - 1/3 Tr(E)} )
+     * For a generic (potentially non-linear) constitutive law, one cannot assume that RHS = F - LHS*current_values.
+     * Because of this, the AddViscousTerm function manages both the LHS and the RHS.
+     */
+    this->AddViscousTerm(rData,LHS,rLocalRHS);
+
+    noalias(rLocalLHS) += LHS;
+}
+
+template< class TElementData >
+void AlternativeDVMSDEMCoupled<TElementData>::AddMassLHS(
+    TElementData& rData,
+    MatrixType &rMassMatrix)
+{
+    const double density = this->GetAtCoordinate(rData.Density,rData.N);
+    const double fluid_fraction = this->GetAtCoordinate(rData.FluidFraction, rData.N);
+    // Note: Dof order is (u,v,[w,]p) for each node
+    for (unsigned int i = 0; i < NumNodes; i++)
+    {
+        unsigned int row = i*BlockSize;
+        for (unsigned int j = 0; j < NumNodes; j++)
+        {
+            unsigned int col = j*BlockSize;
+            const double Mij = rData.Weight * density * fluid_fraction * rData.N[i] * rData.N[j];
+            for (unsigned int d = 0; d < Dim; d++)
+                rMassMatrix(row+d,col+d) += Mij;
+        }
+    }
+
+    /* Note on OSS and full projection: Riccardo says that adding the terms provided by
+     * AddMassStabilization (and incluiding their corresponding terms in the projeciton)
+     * could help reduce the non-linearity of the coupling between projection and u,p
+     * However, leaving them on gives a lot of trouble whith the Bossak scheme:
+     * think that we solve F - (1-alpha)*M*u^(n+1) - alpha*M*u^(n) - K(u^(n+1)) = 0
+     * so the projection of the dynamic terms should be Pi( (1-alpha)*u^(n+1) - alpha*u^(n) )
+     */
+    if ( rData.UseOSS != 1.0 )
+        this->AddMassStabilization(rData,rMassMatrix);
+}
+
+template<class TElementData>
+void AlternativeDVMSDEMCoupled<TElementData>::MassProjTerm(
+    const TElementData& rData,
+    double &rMassRHS) const
+{
+        const auto velocities = rData.Velocity;
+
+        const auto fluid_fraction = this->GetAtCoordinate(rData.FluidFraction, rData.N);
+        const auto mass_source = this->GetAtCoordinate(rData.MassSource, rData.N);
+        const auto fluid_fraction_rate = this->GetAtCoordinate(rData.FluidFractionRate, rData.N);
```

Comment:
```suggestion
        const double fluid_fraction = this->GetAtCoordinate(rData.FluidFraction, rData.N);
        const double mass_source = this->GetAtCoordinate(rData.MassSource, rData.N);
        const double fluid_fraction_rate = this->GetAtCoordinate(rData.FluidFractionRate, rData.N);
```
Maybe I'm wrong but I think that all these are of `double` type.